Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm

```diff
@@ -4123,6 +4123,7 @@ void determineChangedTracksFromNewTracksAndOldItems(MediaSelectionGroupAVFObjC*
     else if (RetainPtr videoTarget = std::exchange(m_videoTarget, nullptr)) {
         INFO_LOG(LOGIDENTIFIER, "Clearing videoTarget");
         [m_avPlayer removeVideoTarget:videoTarget.get()];
+        [m_videoLayer setPlayer:m_avPlayer.get()];
```

Comment:
I'm surprised this is necessary, because we should have restored the player when the endpoint became null.

Reply:
It depends upon the order in which `setVideoTarget()` and `isInFullscreenOrPictureInPictureChanged()` are called. If this method is called first, then `setVideoTarget()` becomes a no-op, and the layer's player is never restored.